src/include/platform/ConnectivityManager.h

```diff
@@ -123,6 +124,9 @@ class ConnectivityManager
 
     struct ThreadPollingConfig;
 
+    struct ThreadDiscoveryResult;
+    typedef void (*ThreadDiscoveryResultCallback)(ThreadDiscoveryResult & aResult);
```

Comment:
why not ByteRange as argument? Reasoning being that data is supposed to be opaque from CHIP perspective